ui/lib/CreateStream.tsx

```diff
@@ -71,60 +162,247 @@ const CreateStream = (props: CreateStreamProps) => {
     functionName: "approve",
     args: [sm.address, selectedTime * props.amountPerSecond],
   });
-  const { write: approveStream } = useContractWrite(approvalConfig);
+
+  const {
+    data,
+    isLoading,
+    // isSuccess: Make sure transaction has been approved by user (used to get tx hash)
+    isSuccess,
+    isError,
+    write: approveStream,
+  } = useContractWrite(approvalConfig);
+
+  // Then make sure transaction has been processed once it has been approved by user
+  const [txHash, setTxHash] = useState<string | undefined>(undefined);
+  useEffect(() => {
+    if (isSuccess && data?.hash) {
+      setTxHash(data?.hash);
+    }
+  }, [isSuccess, data?.hash]);
+
+  const {
+    error: txError,
+    isSuccess: txSuccess,
+    isLoading: txLoading,
+  } = useWaitForTransaction({
+    hash: txHash as `0x${string}`,
```

Comment:
what happems where `txHash` is `undefined`?

Reply:
the logic is done on the previous step; `  useEffect(() => {
    if (isSuccess && data?.hash) {
      setTxHash(data?.hash);
    }
  }, [isSuccess, data?.hash]);
`

subsequent function only loads only with isSuccess; otherwise you get  {isError && <div>You did not confirm the transaction</div>}